src/plugins/intel_cpu/src/transformations/snippets/x64/op/brgemm_copy_b.cpp

```diff
@@ -98,18 +138,29 @@ void intel_cpu::BrgemmCopyB::compute_block_size_values(const size_t blk_size_k,
     m_N_blk = blk_size_n != 0 ? blk_size_n : *input_shape.rbegin();
 }
 
-ov::Shape intel_cpu::BrgemmCopyB::get_repacking_buffer_shape() const {
-    // OneDNN implementation repacks data and always writes the result by m_brgemmVNNIFactor * m_inner_n_block blocks
-    // despite the actual size of the input data. Because of that we have to round-up the allocation shape to always have enough memory allocated.
-    // For the details, please see 'copy_4x64' and 'copy_2x32' implementations and usage in onednn/src/cpu/x64/matmul/brgemm_matmul_copy_utils.cpp
-    return ov::Shape{rnd_up(m_K_blk, m_brgemmVNNIFactor), rnd_up(m_N_blk, m_inner_n_block)};
+size_t intel_cpu::BrgemmCopyB::get_repacking_buffer_size() const {
+    // Repacking buffer shape is set in accordance to OneDNN requirements
+    const size_t N_dim = std::max(m_N_blk, m_inner_n_block);
+    if (with_transpose()) {
+        // In case of transpose, K dimension must be rounded-up to number of elems in vector register
+        // For the details, please see 'transpose16x8' and 'fixup16x16' implementations and usage in onednn/src/cpu/x64/matmul/brgemm_matmul_copy_utils.cpp
+        OPENVINO_ASSERT(dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core), "BrgemmCopyB doesn't support non avx512 platforms");
```

Comment:
Exactly. Do you think we cant move all the onednn-specific low-level details to static emitter functions? Including compute_inner_block and compute_vnni_factor?
Node should not care about particular details of emitter implementations. And if we do need to expose them because they are used in the transformation pipeline, let's at least keep the implementation in the emitter.
Again, if some of these details will be updated (e.g. onednn migration), we will need to make changes only in the emitter. The rest of the pipeline will pick up the changes and work normally.

Reply:
Good idea, I moved emitter specific computations in emitter static functions